### ISSUE TITLE
can't seem to figure this out.

### ISSUE THREAD

**Issue by Psycheux** [SantX27/yu-topy#2]
i might just be stupid but when i run the programme it just tells me that there is no file for xxx.txt (example : yst00140.txt) 
i also tried to make a folder that has both the ybn files and then those same files but instead of .ybn they are .txt and tried it with both .ybn and .txt versions and got another error. I'm just uncertain if i'm screwing something up. this is also the English copy of Euphoria so maybe thats it? i also used YuRISTools since i couldnt find a copy of ToolForYuris, so maybe that's the issue? I'm just curious, this may be a dumb ish thing but i'd love to try getting it working. (also im sorry if my wording is not that great, i'm just dumb with not a ton of experience)
![1](https://github.com/SantX27/yu-topy/assets/118335552/e0039c0f-a017-429c-8d7a-9535d47ca106)
![2](https://github.com/SantX27/yu-topy/assets/118335552/7e6c252d-dbb3-465f-9197-f5e1e3728844)

**Comment by SantX27** [comment]
Hi,
First of all, thank you for checking out this repository, it truly means a lot to us.
Second, we're sorry if you lost time figuring out how the hell to run the converter, we just never thought somebody would actually try and use it, so we never made a proper documentation on how to get everything set up correctly.
As for the errors you are getting, yu-topy can't actually decode ybns, but it uses the txt files that ToolForYuris spits out.
I can't give a proper indepth tutorial rn, but the gist of it is:

- Get a copy of Euphoria HD English (I suppose you've already got the right version, is the one that can be bought from MangaGamer or "borrowed" from IGGames)
- Unpack all files with [ToolForYuris](https://arca.live/b/simya/37085251), I don't remember the flags, but you should end up with all the cgs, sounds and scripts in a standard format (png, ogg, and txt)
- Use yu-topy on the txt files, it should make a corresponding rpy for every txt
- Create a new renpy project, and get inside the new project folder
- Place all the folders containing pngs (bg, cgsys, stand, ...) inside the images folder
- Place all the folders containing oggs (bgm, se, voice) inside the audio folder
- Edit script.rpy so that it contains
```
label start:
    call yu_00_op_01
    return
```
Dump all the rpy files that yu-topy has generated in the same folder script.rpy is in, then you should be able to run the most borked version of euphoria that has ever existed. That if this project even runs on Windows to begin with, I might've actually hardcoded some paths, but I'll be happy to solve them if they pop up.
I will take time to make a proper setup guide, since you showed interest.
You should be able to see the first sex scenes for each character, then everything breaks down because I've never implemented the character selection script, however you can use ren'py built in debugger to drop in whatever rpy you like, so you can technically finish the game.

**Comment by Psycheux** [comment]
nyot to sound like a broken record or be too annoying but do you happen to have a backup of ToolForYuris? 
I cant seem to download from the provided link (i can view the link while in america but used a VPN for Canada and could see it but it seems to still be unavailable, i attatched screencaps) 

(edit: imma still follow the rest of what you said with what i have to see if i can figure something out, but thank you in advance)
![screencap_0_172](https://github.com/SantX27/yu-topy/assets/118335552/0b05ec5a-04ae-4da0-a922-842e9486f95c)

![screencap_1_173](https://github.com/SantX27/yu-topy/assets/118335552/8a760e54-01d3-45f1-8589-4f36e4c8438e)

**Comment by MrPalloncini** [comment]
Apparently we had an old link for the version we used for development (v1.16)
These are the current official links for the 1.21 version, that should be working:
https://arca.live/b/handtranslator/37085669
https://mega.nz/file/YsV3QaDI#Nr2apJiFJibHv1R2vad8uuyIR1ovNthTWsbTUpHXdAU
Also, for posterity, in case the links go down again, we have archived both 1.16 and 1.21:
https://mega.nz/folder/cdBRTZzC#GijdUmHvNtaLc1RJA6vlmw
In case these links go down, please contact me at santikrusso@gmail.com, if I am not dead yet I will reply with a copy of the files.

**Comment by Msfrhdsa** [comment]
What to do with files like yst_list.ybn ysc.ybn yscfg.ybn in euphoria? They also contain some scripts, right? They can't be decoded, and I don't think the game will work normally with endings without them.

**Comment by SantX27** [comment]
You're absolutely correct. yst_list is somewhat decoded by that hellish regex, and it contains just the file name and other trival information regarding all the script files. Some other files, like ysv and ysl are really important, since they contain accordingly the variables and the labels. This "release" of yu-topy can't process these files, it's stuck with whatever ToolForYuris decodes, so it absolutely doesn't know jack ass about jumps, gotos, ifs and whatever else is going on in the logic side of the Yuris scripting language, it can only read text, image and sound data and shove all this jumbled data into Renpy's throat, as of now it's just a testament on the flexibility of Renpy, nothing else.

For the past 4-5 months, I have started, in the few hours I can get without feeling sick, a total rewrite of yu-topy, that doesn't necessitate ToolForYuris in order to work and that is able to correctly parse most of the config files you've cited and all the actual script files. The development it's going to be very long, I am a novice of both python and byte parsing solutions, and most of the code that I end up writing ends up being deleted shortly after since i keep finding "better" ways of doing stuff. I would love to finish what I and my friend started, but I can't find someone, not even myself, that actually cares about getting all of this stuff done.

I will make a new branch with this new code, if someone that is reading this is interested in helping out, in any way, shape or form, I will be glad to be more proactive with the development side of things.
Also, here's the document that inspired me to restart yutopy's development, it contains the "map" of all the bytes inside each yuris file: https://github.com/arcusmaximus/VNTranslationTools/blob/main/VNTextPatch.Shared/Scripts/Yuris/Notes.txt

**Comment by SantX27** [comment]
Update: the new code is up on the pron branch, I need to port the struct syntax to all decoders.

**Comment by SantX27** [comment]
Update: got most of the files decoded properly, the program is now able to parse and decode correctly all of the data structures, files and instructions of Euphoria HD english, the SDK for version 474 and probably even more Yu-ris v474 VNs that follow the encoding conventions, so we can start working on a better Yu-ris to Ren'py converter.
I must add also that, instead of being a logical challenge, decoding all of this stuff was an exercise in stupidity, most of the data structures are a jumbled mess and the resulting instructions are even more illogical, I'm honestly amazed that Yu-ris is even able to work, let alone that is functioning as well as it is.

**Comment by SantX27** [comment]
![image](https://github.com/user-attachments/assets/d3031a27-462b-4930-8671-a9f4ffa33fb5)

**Comment by SantX27** [comment]
It's been a hot while... and after the smooth sail that was decoding the data, actually interpreting it and converting all that crap into Ren'py has been real tough. So much so that I'm actually unsure on how to continue the project.
